src/audio/copier/copier.c

```diff
@@ -1667,10 +1721,37 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 				break;
 			case SOF_COMP_DAI:
 				if (cd->endpoint_num == 1) {
-					ret = dai_zephyr_params(cd->dd[0], cd->endpoint[0],
-								params);
+					if (params->direction == SOF_IPC_STREAM_PLAYBACK) {
+						params->buffer.size = cd->config.base.obs;
+						params->sample_container_bytes =
+							cd->out_fmt->depth / 8;
+						params->sample_valid_bytes =
+							cd->out_fmt->valid_bit_depth / 8;
```

Comment:
that's 6 indentation levels... copier.c becomes quite hard to manage...

Reply:
not sure later multi endpoint can reduce this or not, I also feel it is too long.